recipes/userver/all/conanfile.py

```diff
@@ -0,0 +1,655 @@
+        self.requires('concurrentqueue/1.0.3', transitive_headers=True)
+        self.requires('cryptopp/8.7.0')
+        self.requires('fmt/8.1.1', transitive_headers=True)
+        self.requires('libnghttp2/1.51.0')
+        self.requires('libcurl/7.88.1')
+        self.requires('libev/4.33')
+        self.requires('http_parser/2.9.4')
+        self.requires("openssl/[>=1.1 <4]")
+        self.requires('rapidjson/cci.20220822', transitive_headers=True)
+        self.requires('yaml-cpp/0.7.0')
+        self.requires("zlib/[>=1.2.11 <2]")
+
+        if self.options.with_jemalloc:
+            self.requires('jemalloc/5.3.0')
+        if self.options.with_grpc:
+            self.requires(
+                'grpc/1.54.3', transitive_headers=True, transitive_libs=True,
+            )
+            self.requires(
+                'googleapis/cci.20230501',
+                transitive_headers=True,
+                transitive_libs=True,
+            )
+            self.requires(
+                'grpc-proto/cci.20220627',
+                transitive_headers=True,
+                transitive_libs=True,
+            )
+            self.requires('protobuf/3.21.12') # override
+        if self.options.with_postgresql:
+            self.requires('libpq/14.5')
+        if self.options.with_mongodb:
+            self.requires('cyrus-sasl/2.1.27')
+            self.requires(
+                'mongo-c-driver/1.22.0',
+                transitive_headers=True,
+                transitive_libs=True,
+            )
+        if self.options.with_redis:
+            self.requires('hiredis/1.0.2')
+        if self.options.with_rabbitmq:
+            self.requires('amqp-cpp/4.3.26')
+        if self.options.with_clickhouse:
+            self.requires('clickhouse-cpp/2.4.0')
+            self.requires(
+                'abseil/20230125.3',
+                transitive_headers=True,
+                transitive_libs=True,
+            )
+        if self.options.with_utest:
+            self.requires(
+                'gtest/1.12.1', transitive_headers=True, transitive_libs=True,
+            )
+            self.requires(
+                'benchmark/1.6.2',
+                transitive_headers=True,
+                transitive_libs=True,
+            )
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True,
+                    destination=self.source_folder)
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("userver can't be built on Windows")
+        if self.options.shared:
+            raise ConanInvalidConfiguration("userver can't be built as shared")
+        if self.dependencies['mongo-c-driver'].options.with_sasl != 'cyrus':
+            raise ConanInvalidConfiguration(
+                f'{self.ref} requires mongo-c-driver with_sasl cyrus',
+            )
```

Comment:
Oh I now see that the whole recipe then has no built packages in the case of CCI as everything would be skipped. Not sure what's the best way forward then - as I said we're working on allowing custom configurations to be built, but that's maybe a bit ways off

Reply:
If you need logs, I upstreamed conanfile and now it can be checked in main repo https://github.com/userver-framework/userver but it uses conan v1 due to jemalloc does not support v2. Of course I'll try to upstream your suggestions too.